prow/cmd/checkconfig/main.go

```diff
@@ -134,16 +139,73 @@ func main() {
 	// presence won't lead to strictly incorrect behavior, so we can
 	// detect them here but don't necessarily want to stop config re-load
 	// in all components on their failure.
+	var errs []error
 	if o.warningEnabled(mismatchedTideWarning) {
 		if err := validateTideRequirements(configAgent, pluginAgent); err != nil {
-			o.reportWarning(err, "Invalid tide merge configuration.")
+			errs = append(errs, err)
 		}
 	}
 	if o.warningEnabled(nonDecoratedJobsWarning) {
 		if err := validateDecoratedJobs(configAgent); err != nil {
-			o.reportWarning(err, "Invalid tide merge configuration.")
+			errs = append(errs, err)
+		}
+	}
+	if o.warningEnabled(jobNameLengthWarning) {
+		if err := validateJobRequirements(configAgent); err != nil {
+			errs = append(errs, err)
 		}
 	}
+	if len(errs) > 0 {
+		reportWarning(o.strict, errorutil.NewAggregate(errs...))
+	}
+}
+
+func validateJobRequirements(configAgent config.Agent) error {
+	c := configAgent.Config().JobConfig
+
+	var validationErrs []error
+	for repo, jobs := range c.Presubmits {
+		for _, job := range jobs {
+			validationErrs = append(validationErrs, validatePresubmitJob(repo, job))
+		}
+	}
+	for repo, jobs := range c.Postsubmits {
+		for _, job := range jobs {
+			validationErrs = append(validationErrs, validatePostsubmitJob(repo, job))
+		}
+	}
+	for _, job := range c.Periodics {
+		validationErrs = append(validationErrs, validatePeriodicJob(job))
+	}
+
+	return errorutil.NewAggregate(validationErrs...)
+}
+
+func validatePresubmitJob(repo string, job config.Presubmit) error {
```

Comment:
I think you can DRY this out if you take the ObejctMeta as input instead of the full config

Reply:
@stevekuznetsov I would love to, but I am missing few labels :D